src/Appwrite/Platform/Workers/Migrations.php

```diff
@@ -278,24 +275,43 @@ protected function processMigration(Document $project, Document $migration, Log
                 $destination
             );
 
-            /** Start Transfer */
-            $migrationDocument->setAttribute('stage', 'migrating');
+            // Calculate group resources
+            $resources = $group->getAttribute('resources');
+
+            switch ($group->getAttribute('group')) {
+                case Transfer::GROUP_AUTH:
+                    $resources = array_intersect(Transfer::GROUP_AUTH_RESOURCES, $resources);
+                    break;
+                case Transfer::GROUP_STORAGE:
+                    $resources = array_intersect(Transfer::GROUP_STORAGE_RESOURCES, $resources);
+                    break;
+                case Transfer::GROUP_DATABASES:
+                    $resources = array_intersect(Transfer::GROUP_DATABASES_RESOURCES, $resources);
+                    break;
+                case Transfer::GROUP_FUNCTIONS:
+                    $resources = array_intersect(Transfer::GROUP_FUNCTIONS_RESOURCES, $resources);
+                    break;
+                default:
+                    throw new Exception('Migration worker was initialized with unknown group');
+            }
```

Comment:
You already did an array_intersect in the API level. This doesn't seem to be required